packages/opentelemetry-exporter-collector/src/platform/browser/sendSpans.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as core from '@opentelemetry/core';
+import { Logger } from '@opentelemetry/types';
+import { CollectorExporter } from '../../CollectorExporter';
+import * as collectorTypes from '../../types';
+import { VERSION } from '../../version';
+
+/**
+ * function that is called once when {@link ExporterCollector} is initialised
+ * @param shutdownF shutdown method of {@link ExporterCollector}
+ */
+export function onInit(shutdownF: EventListener) {
+  window.addEventListener('unload', shutdownF);
+}
+
+/**
+ * function to be called once when {@link ExporterCollector} is shutdown
+ * @param shutdownF - shutdown method of {@link ExporterCollector}
+ */
+export function onShutdown(shutdownF: EventListener) {
+  window.removeEventListener('unload', shutdownF);
+}
+
+/**
+ * function to send spans to the [opentelemetry collector]{@link https://github.com/open-telemetry/opentelemetry-collector}
+ *     using the standard http/https node module
+ * @param spans
+ * @param onSuccess
+ * @param onError
+ * @param collectorExporter
+ */
+export function sendSpans(
+  spans: collectorTypes.Span[],
+  onSuccess: () => void,
+  onError: (status?: number) => void,
+  collectorExporter: CollectorExporter
+) {
+  const exportTraceServiceRequest: collectorTypes.ExportTraceServiceRequest = {
+    node: {
+      identifier: {
+        hostName: collectorExporter.hostName || window.location.host,
+        startTimestamp: core.hrTimeToTimeStamp(core.hrTime()),
+      },
+      libraryInfo: {
```

Comment:
It would be nice if you can add unit tests to verify these libraryInfo/versions.